moveit_core/robot_model/src/robot_model.cpp

```diff
@@ -84,6 +84,7 @@ void moveit::core::RobotModel::buildModel(const urdf::ModelInterface &urdf_model
   root_joint_ = NULL;
   root_link_ = NULL;
   link_geometry_count_ = 0;
+  variable_count_ = 0;
```

Comment:
On Mon, Aug 15, 2016 at 12:38:11PM -0700, Dave Coleman wrote:
> I don't think this belongs here - it gets set to 0 later in subfunction ``buildJointInfo()`` when it is used 

No it does not. It is *computed* there (it is incremented for each joint later in that routine).
But importantly, this only happens if there is a root joint.

> This should go in the constructor or (my fav) in the member variable declaration in C++11

You're welcome to refactor the `RobotModel` class.
As of now (and especially in indigo) I would prefer to keep this default initialization together with similar initializations.
That's where I put it, to fix the missing initialization when no root link is present.

Everything else would be a separate request imho.

Also, I'm not sure yet what to make out of C++11's member variable initialization.
To me, it feels better to have the initialization values of member variables with the implementation of the constructor and not with the definition of the class.
Internally, this seems to be what is happening even with the C++11 feature - this can have weird consequences.